plugins/hls-tactics-plugin/src/Wingman/LanguageServer.hs

```diff
@@ -77,18 +83,27 @@ runStaleIde state nfp a = MaybeT $ runIde state $ useWithStale a nfp
 
 
 ------------------------------------------------------------------------------
--- | Get the the plugin config
-getTacticConfig :: MonadLsp Plugin.Config m => ShakeExtras -> m Config
-getTacticConfig extras = do
-  pcfg <- getPluginConfig extras "tactics"
-  pure $ case fromJSON $ Object $ plcConfig pcfg of
-    Success cfg -> cfg
-    Error _     -> emptyConfig
 
+properties :: Properties
+  '[PropertyKey "max_use_ctor_actions" 'TNumber,
+    PropertyKey "features" 'TString]
+properties = emptyProperties
+  & defineStringProperty @"features"
+    "Features set used by wingman (tactic) plugin" ""
+  & defineNumberProperty @"max_use_ctor_actions"
+    "Maximum number of `Use constructor <x>` code actions that can appear" 5
+
+
+-- | Get the the plugin config
+getTacticConfig :: MonadLsp Plugin.Config m => PluginId -> m Config
+getTacticConfig pId =
+  Config
+    <$> (parseFeatureSet <$> usePropertyLsp @"features" pId properties)
+    <*> usePropertyLsp @"max_use_ctor_actions" pId properties
```

Comment:
Previously,  "tactics" was hardcoded as the key of custom config section, which didn't accord with the plugin id ("tactic"). I did a simple migration and unified them :) @isovector

Reply:
It should be "tactics". If the plugin id disagrees, it's the plugin id that's wrong!